src/app/templates/SwapForm/SwapForm.tsx

```diff
@@ -109,15 +110,36 @@ export const SwapForm: FC = () => {
     [inputValue.assetSlug, outputValue.assetSlug]
   );
 
+  const minimumReceivedAmount = useMemo(() => {
+    if (bestTradeWithSlippageTolerance.length > 0) {
+      const lastTradeOperation = bestTradeWithSlippageTolerance[bestTradeWithSlippageTolerance.length - 1];
+
+      const amount = atomsToTokens(lastTradeOperation.bTokenAmount, outputAssetMetadata.decimals);
+
+      return amount;
+    }
+    return new BigNumber(0);
+  }, [bestTradeWithSlippageTolerance, outputAssetMetadata.decimals]);
+
   useEffect(() => {
-    if (inputMutezAmount && routePairsCombinations.length > 0) {
-      const bestTradeExactIn = getBestTradeExactInput(inputMutezAmount, routePairsCombinations);
-      const bestTradeOutput = getTradeOutputAmount(bestTradeExactIn);
+    if (bestTrade) {
+      const bestTradeOutput = getTradeOutputAmount(bestTrade);
 
-      const outputTzAmount = bestTradeOutput ? atomsToTokens(bestTradeOutput, outputAssetMetadata.decimals) : undefined;
+      const outputTzAmount = bestTradeOutput
+        ? atomsToTokens(bestTradeOutput, outputAssetMetadata.decimals)
+        : new BigNumber(0);
 
+      const feeAmount = minimumReceivedAmount.minus(minimumReceivedAmount.multipliedBy(ROUTING_FEE_RATIO));
```

Comment:
inaccurate fee calculation due to rounding (take a look at `getRoutingFeeTransferParams`, how `feeAmount` is calcalated )